app/controllers/concerns/course/achievement_conditional_concern.rb

```diff
@@ -0,0 +1,9 @@
+module Course::AchievementConditionalConcern
+  def return_to_path
+    course_achievements_path(@course)
+  end
+
+  def conditional=
```

Comment:
Isn't this supposed to be some kind of `before_action`? See how cancancan automatically loads resources over routes.